### PR TITLE
Remove paths based on __file__, separate user data from app data

### LINKED ISSUE #3862: Remove all paths based on __file__
Branch: develop
Install method: `python setup.py install --static` within a virtualenv, on Ubuntu

I previously downloaded the EN language pack with no issue, and it worked. Now, I tried downloading the Spanish language pack, and got the following error:

image

It looks like maybe it's trying to unpack or otherwise write something to the **file path** `/static` (i.e. using the URL path as a file path)?

### PR DESCRIPTION
**Pull request by MCGallaspy**
Starts to fix #3862. Going piece by piece. Khan exercise translations now live in user data space, and I tested it out to see that it works. More to come.

### CODE REVIEW

**Comment by MCGallaspy** [comment]
Okay, done here. Other uses of `__file__` appear to be legitimate -- referencing locations within the app that don't need to be user-writable -- or are in apps which are disabled and have no plan for further development (playlist, store).

**Comment by MCGallaspy** [comment]
Fixes #3682.

**Comment by aronasorman** [comment]
KA Lite in general seems to be working for me. However in the course of testing this PR I found another bug related to i18n (switching languages when `not DEBUG` results in a 500 error). I'll hold off on merging until I find the cause of this.

**Comment by aronasorman** [comment]
This is the error from the console:
```python
(snip)
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/templatetags/compress.py", line 149, in render
    return self.render_compressed(context, self.kind, self.mode, forced=forced)
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/templatetags/compress.py", line 102, in render_compressed
    cache_key, cache_content = self.render_cached(compressor, kind, mode, forced=forced)
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/templatetags/compress.py", line 81, in render_cached
    cache_key = get_templatetag_cachekey(compressor, mode, kind)
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/cache.py", line 94, in get_templatetag_cachekey
    "templatetag.%s.%s.%s" % (compressor.cachekey, mode, kind))
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/utils/decorators.py", line 39, in __get__
    value = obj.__dict__[self.__name__] = self.__get(obj)
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/base.py", line 172, in cachekey
    [self.content] + self.mtimes).encode(self.charset), 12)
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/utils/decorators.py", line 39, in __get__
    value = obj.__dict__[self.__name__] = self.__get(obj)
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/base.py", line 166, in mtimes
    for kind, value, basename, elem in self.split_contents()
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/js.py", line 18, in split_contents
    basename = self.get_basename(attribs['src'])
  File "/home/aron/src/virtualenvs/ka-lite/local/lib/python2.7/site-packages/compressor/base.py", line 86, in get_basename
    "compressed" % (url, base_url))
UncompressableFileError: '/content/locale/js/i18n/es.js' isn't accessible via COMPRESS_URL ('/static/') and can't be compressed
```

**Comment by aronasorman** [comment]
Error fixed. I hope you don't mind me pushing to your branch directly.

**Comment by benjaoming** [comment]
> Khan exercise translations now live in user data space, and I tested it out to see that it works. More to come.

Cool, that makes sense.

File: kalite/contentload/management/commands/contentload.py [hunk 1]
```diff
@@ -14,13 +13,8 @@
 from django.core.management.base import NoArgsCommand
 from django.utils.text import slugify
 
-from kalite import topic_tools
-
 from fle_utils.general import ensure_dir
 
-# get the path to an exercise file, so we can check, below, which ones exist
-EXERCISE_FILEPATH_TEMPLATE = os.path.join(settings.KHAN_EXERCISES_DIRPATH, "exercises", "%s")
```
Comment (benjaoming) [inline kalite/contentload/management/commands/contentload.py]: Confirming this was unused (also in other modules). Nice find.

**Comment by benjaoming** [comment]
@MCGallaspy 
Before you go to far, please take care that https://github.com/learningequality/ka-lite/pull/3934 hasn't been merged and already provides a conditional setting for having assessment items as user data if no pre-shipped system-wide location exists.

**Comment by aronasorman** [comment]
Merging now.

**Pull request merged**